auth/auth-server/src/telemetry/helpers.rs

```diff
@@ -0,0 +1,154 @@
+//! Helper methods for capturing telemetry information throughout the auth
+//! server
+
+use crate::error::AuthServerError;
+use crate::telemetry::labels::{
+    ASSET_METRIC_TAG, EXTERNAL_MATCH_BASE_VOLUME, EXTERNAL_MATCH_QUOTE_VOLUME,
+    EXTERNAL_ORDER_BASE_VOLUME, EXTERNAL_ORDER_QUOTE_VOLUME, KEY_DESCRIPTION_METRIC_TAG,
+    NUM_ATOMIC_MATCH_REQUESTS,
+};
+use renegade_api::http::external_match::{
+    AtomicMatchApiBundle, ExternalMatchRequest, ExternalMatchResponse,
+};
+use renegade_circuit_types::order::OrderSide;
+use renegade_common::types::token::Token;
+use renegade_common::types::TimestampedPrice;
+use renegade_util::hex::biguint_to_hex_addr;
+use tracing::warn;
+
+/// Get the human-readable asset and volume of
+/// the given mint and amount.
+/// The asset is the token ticker, if it is found, otherwise
+/// the token's address.
+/// The amount is the decimal amount of the transfer, going through
+/// lossy f64 conversion via the associated number of decimals
+fn get_asset_and_volume(mint: &str, amount: u128) -> (String, f64) {
+    let token = Token::from_addr(mint);
+    let asset = token.get_ticker().unwrap_or(mint.to_string());
+    let volume = token.convert_to_decimal(amount);
+
+    (asset, volume)
+}
+
+/// Calculates the quote per base price from a match bundle
+/// Returns the price as an f64 decimal adjusted value
+fn calculate_quote_per_base_price(match_bundle: &AtomicMatchApiBundle) -> f64 {
+    let (quote, base) = match match_bundle.match_result.direction {
+        OrderSide::Buy => (&match_bundle.send, &match_bundle.receive),
+        OrderSide::Sell => (&match_bundle.receive, &match_bundle.send),
+    };
+
+    Token::from_addr(&quote.mint).convert_to_decimal(quote.amount)
+        / Token::from_addr(&base.mint).convert_to_decimal(base.amount)
```

Comment:
nit: I'd break this up into a few smaller lines:
```rust
let quote_amt = Token::from_addr(&quote.mint).convert_to_decimal(quote.amount);
let base_amt = Token::from_addr(&base.mint).convert_to_decimal(base.amount);
quote_amt / base_amt
```
Slightly more readable to an unfamiliar reader in that the intermediate variable names help annotate the code